calc_app/utils.py

```diff
@@ -0,0 +1,60 @@
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+
+
+def get_month_name(month_number):
+    dct = {
+        1: "January",
+        2: "February",
+        3: "March",
+        4: "April",
+        5: "May",
+        6: "June",
+        7: "July",
+        8: "August",
+        9: "September",
+        10: "October",
+        11: "November",
+        12: "December",
+    }
+    return dct[month_number]
+
+
+def get_months_numbs_and_names():
+    """
+    returns numbers and related month's names
+    """
+    months = {}
+    for i in range(1, 13):
+        months[i] = get_month_name(i)
+    return months
+
+
+def get_graph():
+    """
+    returns graph image
+    """
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    buffer.seek(0)
+    image_png = buffer.getvalue()
+    graph = base64.b64encode(image_png)
+    graph = graph.decode("utf-8")
+    buffer.close()
+    return graph
+
+
+def get_plot(sizes, labels):
+    """
+    builds and returns graph
+    """
+    plt.switch_backend("AGG")
+    plt.figure(figsize=(9, 5))
+    figure, axes = plt.subplots()
+    axes.pie(sizes, labels=labels, autopct="%1.1f%%", startangle=90)
```

Comment:
May we update is somehow to avoid such cases:
![image](https://user-images.githubusercontent.com/72129830/206928864-dd404fa5-cdcd-47bc-a2b0-17a7a45dfc7b.png)

???

Reply:
I don't think so, because it is the responsibility of numpy. We can only control the data we pass into pyplot instance. But for example we can not to show the values with 0 %, but it won't entirely improve the situation. But anyway I'll try to solve this problem

Reply:
I think that it is possible with custom configuration of plot

Reply:
> I think that it is possible with custom configuration of plot

Updated